main.py

```diff
@@ -52,108 +78,205 @@ def open_browser():
+                ".css-1b2j8lx"
+            ))
+        ).click()
+
+        # Click again the purchase button
+        logger.debug('find and clicking again the purchase button')
+        WebDriverWait(browser, TIMEOUT).until(
+            EC.visibility_of_element_located((
+                By.XPATH,
+                "//button[@data-testid='purchase-cta-button']"
+            ))
+        ).click()
+
+    except TimeoutException:
+        logger.debug("no license agreement found")
+
     # wait until its visible and then click the purchase button
     logger.debug('find and click on the last purchase button')
     WebDriverWait(browser, TIMEOUT).until(
-        EC.visibility_of_element_located((By.XPATH, "//button[contains(@class,'btn-primary')]"))
+        EC.visibility_of_element_located((
+            By.XPATH,
+            "//button[contains(@class,'btn-primary')]"
+        ))
     ).click()
 
     # wait until its visible and then click the 'I Agree" popup
     # 'Refund and Right of Withdrawal Information' popup
     try:
         logger.debug('accept the conditions of refund popup')
         WebDriverWait(browser, TIMEOUT).until(
-            EC.visibility_of_all_elements_located(
-                (By.XPATH, "//button[contains(@class,'btn-primary')]"))
+            EC.visibility_of_all_elements_located((
+                By.XPATH,
+                "//button[contains(@class,'btn-primary')]"
+            ))
         )[1].click()
-    except (NoSuchElementException, TimeoutException, LookupError) as ex:
+
+    except (NoSuchElementException, TimeoutException, LookupError):
         logger.debug('no refund conditions popup to accept')
 
     logger.debug('Wait for confirmation that checkout is complete')
 
     # Purchase should be complete. Checking for confirmation
-    WebDriverWait(browser, TIMEOUT).until(EC.visibility_of_element_located((
-        By.XPATH,
-        "//h1/span[contains(text(),'Install')]|//span[contains(text(),'Thank you for buying')]"
-    )))
-
+    WebDriverWait(browser, TIMEOUT).until(
+        EC.visibility_of_element_located((
+            By.XPATH,
+            "//h1/span[contains(text(),'Install')]|"
+            "//span[contains(text(),'Thank you for buying')]"
+        ))
+    )
 
-def execute():
-    browser = open_browser()
-    browser.get(FREE_GAMES_PAGE_URL)
 
+def login(browser):
     try:
         logger.debug('find and click on login button')
         el = WebDriverWait(browser, TIMEOUT).until(
-            EC.element_to_be_clickable((By.XPATH, "//a[@href='/login']"))
+            EC.element_to_be_clickable((
+                By.XPATH,
+                "//div[@id='user']/ul/li/a/span"
+            ))
+        )
+        el.click()
+
+        logger.debug('find and click on EpicGame login method')
+        el = WebDriverWait(browser, TIMEOUT).until(
+            EC.element_to_be_clickable((
+                By.XPATH,
+                "//div[@id='login-with-epic']/div[2]/span/h6"
```

Comment:
Following the previous commend here. Tested without the tags following the ID selector and it works the same. It's better to remove to you don't rely on both having the same ID and also being nested under the same elements ul/li/a/span... the ID is unique already and you can solely rely on it

Reply:
Ok, replaced them